diff --git a/lib/pjsip/mac_x86-64/pjproject-2.10/pjmedia/include/pjmedia/stream.h b/lib/pjsip/mac_x86-64/pjproject-2.10/pjmedia/include/pjmedia/stream.h
index 2f74b52..a265ebc 100644
--- a/lib/pjsip/mac_x86-64/pjproject-2.10/pjmedia/include/pjmedia/stream.h
+++ b/lib/pjsip/mac_x86-64/pjproject-2.10/pjmedia/include/pjmedia/stream.h
@@ -458,6 +458,20 @@ pjmedia_stream_get_rtp_session_info(pjmedia_stream *stream,
 			           pjmedia_stream_rtp_sess_info *session_info);
 
 
+/**
+ * Set the Jitter-Buffer to fixed delay mode.
+ *
+ * @param stream	The media stream.
+ *
+ * @param prefetch	The fixed delay value, in ms.
+ *
+ * @return		PJ_SUCCESS on success.
+ */
+PJ_DECL(pj_status_t)
+pjmedia_stream_jbuf_set_fixed(pjmedia_stream *stream,
+							  unsigned prefetch);
+
+
 /**
  * @}
  */
