diff --git a/pjsip/linux_x86-64/pjproject-2.10/pjmedia/src/pjmedia/splitcomb.c b/pjsip/linux_x86-64/pjproject-2.10/pjmedia/src/pjmedia/splitcomb.c
index 300886b..3f35b62 100644
--- a/pjsip/linux_x86-64/pjproject-2.10/pjmedia/src/pjmedia/splitcomb.c
+++ b/pjsip/linux_x86-64/pjproject-2.10/pjmedia/src/pjmedia/splitcomb.c
@@ -31,7 +31,7 @@
 #define TMP_SAMP_TYPE	    pj_int16_t
 
 /* Maximum number of channels. */
-#define MAX_CHANNELS	    16
+#define MAX_CHANNELS	    64
 
 /* Maximum number of buffers to be accommodated by delaybuf */
 #define MAX_BUF_CNT	    PJMEDIA_SOUND_BUFFER_COUNT
