diff --git a/lib/pjsip/mac_x86-64/pjproject-2.10/pjmedia/src/pjmedia/stream.c b/lib/pjsip/mac_x86-64/pjproject-2.10/pjmedia/src/pjmedia/stream.c
index 42942fc..1b43aba 100644
--- a/lib/pjsip/mac_x86-64/pjproject-2.10/pjmedia/src/pjmedia/stream.c
+++ b/lib/pjsip/mac_x86-64/pjproject-2.10/pjmedia/src/pjmedia/stream.c
@@ -3305,3 +3305,13 @@ pjmedia_stream_get_rtp_session_info(pjmedia_stream *stream,
     session_info->rtcp = &stream->rtcp;
     return PJ_SUCCESS;
 }
+
+/**
+ * Set JitterBuffer to fixed Mode
+ */
+PJ_DEF(pj_status_t)
+pjmedia_stream_jbuf_set_fixed(pjmedia_stream *stream,
+							  unsigned prefetch)
+{
+	return pjmedia_jbuf_set_fixed(stream->jb, prefetch / stream->dec_ptime);
+}
